virtcontainers/container.go

```diff
@@ -1267,6 +1267,56 @@ func (c *Container) updateResources(oldResources, newResources ContainerResource
 		// recalculate the actual number of vCPUs if a different number of vCPUs was removed
 		newResources.VCPUs = oldVCPUs - vcpusRemoved
 	}
+	return nil
+}
+
+func (c *Container) updateMemoryResources(oldResources, newResources ContainerResources) error {
+	oldMem := oldResources.Mem
+	newMem := newResources.Mem
+
+	if newMem == 0 || oldMem == newMem {
+		c.Logger().WithFields(logrus.Fields{
+			"old-mem": fmt.Sprintf("%dMB", oldMem),
+			"new-mem": fmt.Sprintf("%dMB", newMem),
+		}).Debug("the actual number of Mem will not be modified")
+		return nil
+	}
+
+	if oldMem < newMem {
+		// hot add memory
+		addMemMB := newMem - oldMem
+		// hot add memory must be aligned to 128MB
+		if addMemMB%128 != 0 {
+			return fmt.Errorf("can not hot add memory which isn't aligned to 128MB")
+		}
+
+		virtLog.Debugf("hot adding %dMB mem", addMemMB)
+		addMemDevice := &memoryDevice{
+			sizeMB: int(addMemMB),
+		}
+		_, err := c.sandbox.hypervisor.hotplugAddDevice(addMemDevice, memoryDev)
+		if err != nil {
+			return err
+		}
+		newResources.Mem = newMem
+		if err := c.sandbox.agent.onlineCPUMem(0); err != nil {
+			return err
+		}
+	} else {
+		// hot remove memory
+		return fmt.Errorf("can't hot remove memory, we don't support this feature yet")
```

Comment:
We do have memory hotplug on Z :smile:  How about this for the error message : ` Memory hotplug unsupported` feels much more concise.

Reply:
Do you mean `Memory hot remove unsupported` ? I think it would be a good choice.

Reply:
No I think hotplug would be the better choice here. I say this because, almost all of IBM Z docs use plug/unplug and even ovirt calls it [hotplug](https://ovirt.org/develop/release-management/features/virt/hot-plug-memory/). Looks like Vmware is the only one calling it hotadd. But I am okay with hot add if the community agrees.